common/src/main/java/com/netflix/conductor/common/run/Workflow.java

```diff
@@ -142,13 +146,24 @@ public void setWorkflowId(String workflowId) {
 	public List<Task> getTasks() {
 		return tasks;
 	}
+
 	/**
-	 * @param tasks the tasks to set
+	 * This method sets the list of tasks that belong to the workflow and also updates the
+	 * {@link #failedTaskId} in case of any task that has a status of {@link com.netflix.conductor.common.metadata.tasks.Task.Status#FAILED}
+	 *
+	 * @param tasks: The list of the tasks that belong to the workflow
 	 */
 	public void setTasks(List<Task> tasks) {
 		this.tasks = tasks;
+		if (Objects.nonNull(tasks)) {
+			this.failedTaskId = tasks.stream()
+					.filter(task -> Task.Status.FAILED.equals(task.getStatus()))
```

Comment:
Do we need to worry about other statuses: CANCELED, COMPLETED_WITH_ERRORS, TIMED_OUT?

Reply:
@vikramsingh2104 Do we need to worry about states other than `Task.Status.FAILED`